doc/snippets/Microsoft.Data.SqlClient/SqlAuthenticationProvider.xml

```diff
@@ -23,33 +23,34 @@
               {
                   public override async Task&lt;SqlAuthenticationToken&gt; AcquireTokenAsync(SqlAuthenticationParameters parameters)
                   {
-                      string clientId = "my-client-id";
-                      string clientName = "My Application Name";
-                      string s_defaultScopeSuffix = "/.default";
-          
-                      string[] scopes = new string[] { parameters.Resource.EndsWith(s_defaultScopeSuffix) ? parameters.Resource : parameters.Resource + s_defaultScopeSuffix };
-          
-                      IPublicClientApplication app = PublicClientApplicationBuilder.Create(clientId)
+                      const string ClientId = "my-client-id";
+                      const string ClientName = "My Application Name";
+                      const string DefaultScopeSuffix = "/.default";
+
+                      string[] scopes = [ parameters.Resource.EndsWith(DefaultScopeSuffix) ? parameters.Resource : parameters.Resource + DefaultScopeSuffix ];
+
+                      IPublicClientApplication app = PublicClientApplicationBuilder.Create(ClientId)
                           .WithAuthority(parameters.Authority)
-                          .WithClientName(clientName)
+                          .WithClientName(ClientName)
                           .WithRedirectUri("https://login.microsoftonline.com/common/oauth2/nativeclient")
                           .Build();
-          
-                      AuthenticationResult result = await app.AcquireTokenWithDeviceCode(scopes,
-                      deviceCodeResult => CustomDeviceFlowCallback(deviceCodeResult)).ExecuteAsync();
+                      using CancellationTokenSource connectionTimeoutCancellation = new CancellationTokenSource(TimeSpan.FromSeconds(parameters.ConnectionTimeout));
+
+                      AuthenticationResult result = await app.AcquireTokenWithDeviceCode(scopes, CustomDeviceFlowCallback)
+                          .ExecuteAsync(connectionTimeoutCancellation.Token);
```

Comment:
I'm in two minds on this. It highlights the point that the connection timeout needs to be a factor on the token acquisition, but `SqlAuthenticationProvider.AcquireTokenAsync` doesn't provide a CancellationToken for us to use which means that this isn't technically the correct value. That could result in odd errors in high-latency environments, where a few seconds of the connection timeout might already have been consumed during the PRELOGIN phase.